--- a/src/datatype.c
+++ b/src/datatype.c
@@ -31,6 +31,11 @@
 
 #include <netinet/ip_icmp.h>
 
+/* ICMPV6_POLICY_FAIL && ICMPV6_REJECT_ROUTE since Linux 3.2.52  */
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
 static const struct datatype *datatypes[TYPE_MAX + 1] = {
 	[TYPE_INVALID]		= &invalid_type,
 	[TYPE_VERDICT]		= &verdict_type,
@@ -874,8 +879,11 @@ static const struct symbol_table icmpv6_
 		SYMBOL("admin-prohibited",	ICMPV6_ADM_PROHIBITED),
 		SYMBOL("addr-unreachable",	ICMPV6_ADDR_UNREACH),
 		SYMBOL("port-unreachable",	ICMPV6_PORT_UNREACH),
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,52)
 		SYMBOL("policy-fail",		ICMPV6_POLICY_FAIL),
 		SYMBOL("reject-route",		ICMPV6_REJECT_ROUTE),
+#endif
+/* ICMPV6_POLICY_FAIL && ICMPV6_REJECT_ROUTE since Linux 3.2.52 */
 		SYMBOL_LIST_END
 	},
 };
