--- a/shell/ash.c
+++ b/shell/ash.c
@@ -13158,7 +13158,7 @@
 		const char *hp;
 
 		state = 1;
-		read_profile("/etc/profile");
+		read_profile("/opt/etc/profile");
  state1:
 		state = 2;
 		hp = lookupvar("HOME");
--- a/shell/hush.c
+++ b/shell/hush.c
@@ -8032,8 +8032,8 @@
 	/* If we are login shell... */
 	if (flags & OPT_login) {
 		FILE *input;
-		debug_printf("sourcing /etc/profile\n");
-		input = fopen_for_read("/etc/profile");
+		debug_printf("sourcing /opt/etc/profile\n");
+		input = fopen_for_read("/opt/etc/profile");
 		if (input != NULL) {
 			close_on_exec_on(fileno(input));
 			install_special_sighandlers();
--- a/loginutils/adduser.c
+++ b/loginutils/adduser.c
@@ -190,7 +190,7 @@
 	die_if_bad_username(pw.pw_name);
 	if (!pw.pw_dir) {
 		/* create string for $HOME if not specified already */
-		pw.pw_dir = xasprintf("/home/%s", argv[0]);
+		pw.pw_dir = xasprintf("/opt/home/%s", argv[0]);
 	}
 	pw.pw_passwd = (char *)"x";
 	if (opts & OPT_SYSTEM_ACCOUNT) {
