--- a/shell/ash.c
+++ b/shell/ash.c
@@ -14347,7 +14347,7 @@ int ash_main(int argc UNUSED_PARAM, char
 		const char *hp;
 
 		state = 1;
-		read_profile("/etc/profile");
+		read_profile("/opt/etc/profile");
  state1:
 		state = 2;
 		hp = lookupvar("HOME");
--- a/shell/hush.c
+++ b/shell/hush.c
@@ -10139,8 +10139,8 @@ int hush_main(int argc, char **argv)
 	/* If we are login shell... */
 	if (flags & OPT_login) {
 		HFILE *input;
-		debug_printf("sourcing /etc/profile\n");
-		input = hfopen("/etc/profile");
+		debug_printf("sourcing /opt/etc/profile\n");
+		input = hfopen("/opt/etc/profile");
 		if (input != NULL) {
 			install_special_sighandlers();
 			parse_and_run_file(input);
--- a/loginutils/adduser.c
+++ b/loginutils/adduser.c
@@ -227,7 +227,7 @@ int adduser_main(int argc UNUSED_PARAM,
 	die_if_bad_username(pw.pw_name);
 	if (!pw.pw_dir) {
 		/* create string for $HOME if not specified already */
-		pw.pw_dir = xasprintf("/home/%s", argv[0]);
+		pw.pw_dir = xasprintf("/opt/home/%s", argv[0]);
 	}
 	pw.pw_passwd = (char *)"x";
 	if (opts & OPT_SYSTEM_ACCOUNT) {
