--- a/include/libbb.h
+++ b/include/libbb.h
@@ -123,16 +123,16 @@
 #endif
 /* Just in case libc doesn't define some of these... */
 #ifndef _PATH_PASSWD
-#define _PATH_PASSWD  "/etc/passwd"
+#define _PATH_PASSWD  "/opt/etc/passwd"
 #endif
 #ifndef _PATH_GROUP
-#define _PATH_GROUP   "/etc/group"
+#define _PATH_GROUP   "/opt/etc/group"
 #endif
 #ifndef _PATH_SHADOW
-#define _PATH_SHADOW  "/etc/shadow"
+#define _PATH_SHADOW  "/opt/etc/shadow"
 #endif
 #ifndef _PATH_GSHADOW
-#define _PATH_GSHADOW "/etc/gshadow"
+#define _PATH_GSHADOW "/opt/etc/gshadow"
 #endif
 #if defined __FreeBSD__ || defined __OpenBSD__
 # include <netinet/in.h>
--- a/loginutils/adduser.c
+++ b/loginutils/adduser.c
@@ -204,7 +204,7 @@
 	char *p;
 	unsigned opts;
 	char *uid;
-	const char *skel = "/etc/skel";
+	const char *skel = "/opt/etc/skel";
 
 #if ENABLE_FEATURE_ADDUSER_LONG_OPTIONS
 	applet_long_options = adduser_longopts;
