--- a/ldso/inlcude/dl-defs.h
+++ b/ldso/include/dl-defs.h
@@ -36,7 +36,7 @@
 #ifndef __LDSO_BASE_FILENAME__
 #define __LDSO_BASE_FILENAME__ "ld.so"
 #endif
-#define LDSO_BASE_PATH UCLIBC_RUNTIME_PREFIX "etc/" __LDSO_BASE_FILENAME__
+#define LDSO_BASE_PATH UCLIBC_RUNTIME_PREFIX "opt/etc/" __LDSO_BASE_FILENAME__
 
 #ifdef __LDSO_PRELOAD_FILE_SUPPORT__
 #define LDSO_PRELOAD LDSO_BASE_PATH ".preload"
--- a/ldso/ldso/dl-elf.c
+++ b/ldso/ldso/dl-elf.c
@@ -319,10 +319,9 @@
 	   This list must exactly match the list in uClibc/ldso/util/ldd.c */
 	_dl_if_debug_dprint("\tsearching full lib path list\n");
 	tpnt1 = search_for_named_library(libname, rflags,
-					UCLIBC_RUNTIME_PREFIX "lib:"
-					UCLIBC_RUNTIME_PREFIX "usr/lib"
+					"/opt/lib:"
 #ifndef __LDSO_CACHE_SUPPORT__
-					":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"
+					":/opt/X11R6/lib"
 #endif
 					, rpnt, NULL);
 	if (tpnt1 != NULL)
