From 8732b3c207362cb952bd9b1412acde62cb146013 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sun, 11 Feb 2024 21:42:09 +0300
Subject: [PATCH] entware-release: bump to 2024.02

---
 entware-release/Makefile | 59 ++++++++++++++++++++--------------------
 1 file changed, 29 insertions(+), 30 deletions(-)

diff --git a/entware-release/Makefile b/entware-release/Makefile
index a71f19d..88808db 100644
--- a/entware-release/Makefile
+++ b/entware-release/Makefile
@@ -1,20 +1,15 @@
+# SPDX-License-Identifier: GPL-2.0-only
 #
-# Copyright (C) 2017-2021 Entware
-#
-# This is free software, licensed under the GNU General Public License v2.
-# See /LICENSE for more information.
-#
-# This is the same package as entware-opt, it does not install busybox. It makes several symlinks instead of it!
-# /opt/... folder order in /opt/etc/profile is different, /opt folders are the last in PATH
-# This package is a replacement of entware-opt for asuswrt firmware
+# Copyright (C) 2017-2024 Entware
 
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=entware-release
-PKG_VERSION:=1.0
-PKG_RELEASE:=2
+PKG_VERSION:=2024.02
+PKG_RELEASE:=1
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/kernel.mk
 
 define Package/entware-release
 	SECTION:=base
@@ -28,9 +23,6 @@ define Package/entware-release/description
  entware_release is a file in /opt/etc with entware info.
 endef
 
-define Build/Prepare
-endef
-
 CPUTYPE=$(if $(CONFIG_CPU_TYPE),$(CONFIG_CPU_TYPE),unknown)
 CPUSUBTYPE=$(if $(CONFIG_CPU_SUBTYPE),$(CONFIG_CPU_SUBTYPE),unknown)
 FLOAT=$(if $(CONFIG_SOFT_FLOAT),soft,hard)
@@ -44,26 +36,33 @@ ifeq ($(CONFIG_SOFT_FLOAT),y)
     endif
 endif
 
-LIBCTYPE=$(if $(CONFIG_LIBC_USE_GLIBC),glibc,uclibc-ng)
-
-define Build/Configure
-	echo "release=entware" > $(PKG_BUILD_DIR)/entware_release
-	echo "arch=$(ARCH)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "cpu=$(CPUTYPE)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "cpu_subtype=$(CPUSUBTYPE)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "float=$(FLOAT)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "gcc=$(call qstrip,$(CONFIG_GCC_VERSION))" >> $(PKG_BUILD_DIR)/entware_release
-	echo "gcc_flags=$(ENVCFLAGS)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "libc=$(LIBCTYPE)" >> $(PKG_BUILD_DIR)/entware_release
-	echo "libc_version=$(call qstrip,$(CONFIG_GLIBC_VERSION))" >> $(PKG_BUILD_DIR)/entware_release
-endef
-
-define Build/Compile
-endef
+# skip steps
+Build/Prepare:=:
+Build/Configure:=:
+Build/Compile:=:
 
 define Package/entware-release/install
 	$(INSTALL_DIR) $(1)/opt/etc
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/entware_release $(1)/opt/etc
+	echo "NAME=\"Entware\"" > $(1)/opt/etc/entware_release
+	echo "ID=entware" >> $(1)/opt/etc/entware_release
+	echo "PRETTY_NAME=\"Entware $(PKG_VERSION)\"" >> $(1)/opt/etc/entware_release
+	echo "VERSION=\"$(PKG_VERSION)\"" >> $(1)/opt/etc/entware_release
+	echo "VERSION_ID=\"$(PKG_VERSION) (sync with OpenWrt)\"" >> $(1)/opt/etc/entware_release
+	echo "HOME_URL=\"https://entware.net/\"" >> $(1)/opt/etc/entware_release
+	echo "SUPPORT_URL=\"https://github.com/Entware/Entware/discussions/\"" >> $(1)/opt/etc/entware_release
+	echo "BUG_REPORT_URL=\"https://github.com/Entware/Entware/issues/\"" >> $(1)/opt/etc/entware_release
+	echo "" >> $(1)/opt/etc/entware_release
+	echo "Build summary:" >> $(1)/opt/etc/entware_release
+	echo "Linux Kernel $(LINUX_VERSION)" >> $(1)/opt/etc/entware_release
+	echo "arch=$(ARCH)" >> $(1)/opt/etc/entware_release
+	echo "cpu_type=$(CPUTYPE)" >> $(1)/opt/etc/entware_release
+	echo "cpu_subtype=$(CPUSUBTYPE)" >> $(1)/opt/etc/entware_release
+	echo "float=$(FLOAT)" >> $(1)/opt/etc/entware_release
+	echo "binutils_version=\"$(call qstrip,$(CONFIG_BINUTILS_VERSION))\"" >> $(1)/opt/etc/entware_release
+	echo "gcc_version=\"$(call qstrip,$(CONFIG_GCC_VERSION))\"" >> $(1)/opt/etc/entware_release
+	echo "gcc_flags=\"$(ENVCFLAGS)\"" >> $(1)/opt/etc/entware_release
+	echo "libc_type=glibc" >> $(1)/opt/etc/entware_release
+	echo "libc_version=\"$(call qstrip,$(CONFIG_GLIBC_VERSION))\"" >> $(1)/opt/etc/entware_release
 endef
 
 $(eval $(call BuildPackage,entware-release))
-- 
2.30.2

