From 6b2cb184a618ff84567801b6c1d7e3bc41bee548 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sun, 11 Feb 2024 18:57:41 +0300
Subject: [PATCH 06/39] include/target.mk: fix post-sync, 2024.02

экстрим (можно ненужно)
---
 include/target.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/target.mk b/include/target.mk
index abb9fdd1..9175f52b 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -202,6 +202,7 @@ ifeq ($(DUMP),1)
     endif
     CPU_CFLAGS += -mno-branch-likely
     CPU_CFLAGS_mips32 = -mips32 -mtune=mips32
+    CPU_CFLAGS_mips32r2 = -mips32r2
     CPU_CFLAGS_mips64 = -mips64 -mtune=mips64 -mabi=64
     CPU_CFLAGS_mips64r2 = -mips64r2 -mtune=mips64r2 -mabi=64
     CPU_CFLAGS_4kec = -mips32r2 -mtune=4kec
-- 
2.30.2

