From 484ebc75ebb7fdec392b8488efb90dc7c602d2f5 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sun, 11 Feb 2024 18:57:41 +0300
Subject: [PATCH 07/39] kselftests-bpf: fix post-sync, 2024.02

изменено
---
 package/devel/kselftests-bpf/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/devel/kselftests-bpf/Makefile b/package/devel/kselftests-bpf/Makefile
index ae845486..0a5b874e 100644
--- a/package/devel/kselftests-bpf/Makefile
+++ b/package/devel/kselftests-bpf/Makefile
@@ -42,7 +42,7 @@ MAKE_VARS = \
 	CROSS_COMPILE="$(TARGET_CROSS)" \
 	SAN_CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS)" \
 	LDLIBS="$(TARGET_LDFLAGS)" \
-	TOOLCHAIN_INCLUDE="$(TOOLCHAIN_DIR)/include" \
+	TOOLCHAIN_INCLUDE="$(TOOLCHAIN_INC_DIRS)" \
 	VMLINUX_BTF="$(LINUX_DIR)/vmlinux"
 
 MAKE_FLAGS = \
-- 
2.30.2

