From 16a1d561436c3ac006863cd446c8b028d0872820 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sun, 11 Feb 2024 18:57:44 +0300
Subject: [PATCH 31/39] target/toolchain: fix post-sync, 2024.02

косметика
---
 target/toolchain/files/wrapper.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/target/toolchain/files/wrapper.sh b/target/toolchain/files/wrapper.sh
index 4b3f8f91..09a9cac9 100755
--- a/target/toolchain/files/wrapper.sh
+++ b/target/toolchain/files/wrapper.sh
@@ -43,6 +43,7 @@ TOOLCHAIN_BIN_DIR="$REALNAME_DIR/"
 
 # Set the PATH so that our run-time location is first
 # (get_feature is run from the path, so this has to be set)
+export ORIG_PATH=${ORIG_PATH:-$PATH}
 export PATH="$TOOLCHAIN_BIN_DIR":$PATH
 export GCC_HONOUR_COPTS
 
-- 
2.30.2

