--- a/chat/Makefile.linux
+++ b/chat/Makefile.linux
@@ -18,7 +18,7 @@
 all:	chat
 
 chat:	chat.o
-	$(CC) -o chat chat.o
+	$(CC) -o chat $(LDFLAGS) chat.o
 
 chat.o:	chat.c
 	$(CC) -c $(CFLAGS) -o chat.o chat.c
--- a/pppdump/Makefile.linux
+++ b/pppdump/Makefile.linux
@@ -11,7 +11,7 @@
 all:	pppdump
 
 pppdump: $(OBJS)
-	$(CC) -o pppdump $(OBJS)
+	$(CC) -o pppdump $(LDFLAGS) $(OBJS)
 
 clean:
 	rm -f pppdump $(OBJS) *~
--- a/pppstats/Makefile.linux
+++ b/pppstats/Makefile.linux
@@ -26,7 +26,7 @@
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
-	$(CC) $(CFLAGS) -o pppstats pppstats.c $(LIBS)
+	$(CC) $(CFLAGS) -o pppstats $(LDFLAGS) pppstats.c $(LIBS)
 
 clean:
 	rm -f pppstats *~ #* core
