--- a/pppd/Makefile.linux
+++ b/pppd/Makefile.linux
@@ -191,8 +191,8 @@
 
 ifdef FILTER
 ifneq ($(wildcard /usr/include/pcap-bpf.h),)
-LIBS    += -lpcap -L$(STAGING_DIR)/usr/lib
-CFLAGS  += -DPPP_FILTER -I$(STAGING_DIR)/usr/include
+LIBS    += -lpcap -L$(STAGING_DIR)/opt/lib
+CFLAGS  += -DPPP_FILTER -I$(STAGING_DIR)/opt/include
 endif
 endif
 
@@ -200,8 +200,8 @@
 PPPDSRCS += pcap_pcc.c
 HEADERS  += pcap_pcc.h
 PPPDOBJS += pcap_pcc.o
-LIBS	+= $(STAGING_DIR)/usr/lib/libpcap.a
-CFLAGS	+= -DPPP_FILTER -DPPP_PRECOMPILED_FILTER -I$(STAGING_DIR)/usr/include
+LIBS	+= $(STAGING_DIR)/opt/lib/libpcap.a
+CFLAGS	+= -DPPP_FILTER -DPPP_PRECOMPILED_FILTER -I$(STAGING_DIR)/opt/include
 endif
 
 ifdef HAVE_INET6
