--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -20,17 +20,17 @@
 #undef _PATH_DEFPATH
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/usr/bin"
 #else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/bin:/usr/bin"
 #endif
 
 #undef _PATH_DEFPATH_ROOT
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
 #else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
 #endif
 
 #define	_PATH_HUSHLOGIN		".hushlogin"
@@ -57,11 +57,11 @@
 #define _PATH_TERMCOLORS_DIR	"/etc/" _PATH_TERMCOLORS_DIRNAME
 
 /* login paths */
-#define _PATH_PASSWD		"/etc/passwd"
-#define _PATH_GSHADOW		"/etc/gshadow"
-#define _PATH_GROUP		"/etc/group"
-#define _PATH_SHADOW_PASSWD	"/etc/shadow"
-#define _PATH_SHELLS		"/etc/shells"
+#define _PATH_PASSWD		"/opt/etc/passwd"
+#define _PATH_GSHADOW		"/opt/etc/gshadow"
+#define _PATH_GROUP		"/opt/etc/group"
+#define _PATH_SHADOW_PASSWD	"/opt/etc/shadow"
+#define _PATH_SHELLS		"/opt/etc/shells"
 
 #ifndef _PATH_TMP
 # define _PATH_TMP		"/tmp/"
### cat 601-fix-paths.patch 
--- a/meson.build
+++ b/meson.build
@@ -20,12 +20,12 @@ sbindir = join_paths(prefixdir, get_opti
 sysconfstaticdir = join_paths(prefixdir, 'lib')
 docdir = join_paths(prefixdir, get_option('datadir'), 'doc', 'util-linux')
 mandir = join_paths(prefixdir, get_option('mandir'))
-runstatedir = '/run'
-execprefixdir = '/usr'
+runstatedir = '/opt/var/run'
+execprefixdir = '/opt'
 usrbin_exec_dir = join_paths(execprefixdir, 'bin')
 usrsbin_exec_dir = join_paths(execprefixdir, 'sbin')
-sysconfdir = '/etc'
-localstatedir = '/var'
+sysconfdir = '/opt/etc'
+localstatedir = '/opt/var'
 bash_completion = dependency('bash-completion', required : get_option('build-bash-completion'))
 
 vendordir = get_option('vendordir')
### cat 602-fix-for-old-kernels.patch 
# XXX <linux/nsfs.h> since Linux 4.9

--- a/config.h.in
+++ b/config.h.in
@@ -313,7 +313,8 @@
 #undef HAVE_LINUX_NET_NAMESPACE_H
 
 /* Define to 1 if you have the <linux/nsfs.h> header file. */
-#undef HAVE_LINUX_NSFS_H
+/* NS_GET_NSTYPE NS_GET_OWNER_UID since Linux 4.11 */
+/* #undef HAVE_LINUX_NSFS_H */
 
 /* Define to 1 if you have the <linux/raw.h> header file. */
 #undef HAVE_LINUX_RAW_H
--- a/meson.build
+++ b/meson.build
@@ -160,7 +160,6 @@ headers = '''
    linux/fiemap.h
    linux/gsmmux.h
         linux/net_namespace.h
-        linux/nsfs.h
         linux/securebits.h
         linux/tiocl.h
         linux/version.h
### cat 610-no-blkzone.patch 
# XXX <linux/blkzoned.h> since Linux 4.10

--- a/meson.build
+++ b/meson.build
@@ -151,7 +151,6 @@ headers = '''
         utmp.h
         utmpx.h
         asm/io.h
-        linux/blkzoned.h
         linux/capability.h
         linux/cdrom.h
         linux/compiler.h
@@ -1353,16 +1352,16 @@ exes += exe
 manadocs += ['sys-utils/blkdiscard.8.adoc']
 bashcompletions += ['blkdiscard']
 
-exe = executable(
-  'blkzone',
-  blkzone_sources,
-  include_directories : includes,
-  link_with : [lib_common],
-  install_dir : sbindir,
-  install : true)
-exes += exe
-manadocs += ['sys-utils/blkzone.8.adoc']
-bashcompletions += ['blkzone']
+#exe = executable(
+#  'blkzone',
+#  blkzone_sources,
+#  include_directories : includes,
+#  link_with : [lib_common],
+#  install_dir : sbindir,
+#  install : true)
+#exes += exe
+#manadocs += ['sys-utils/blkzone.8.adoc']
+#bashcompletions += ['blkzone']
 
 exe = executable(
   'ldattach',
### cat 620-no-lsfd.patch 
# XXX <linux/kcmp.h> since Linux 3.5 + missing hdr for aarch64

--- a/meson.build
+++ b/meson.build
@@ -2447,18 +2447,18 @@ if not is_disabler(exe)
   bashcompletions += ['lsblk']
 endif
 
-exe = executable(
-  'lsfd',
-  lsfd_sources,
-  include_directories : includes,
-  link_with : [lib_common,
-               lib_smartcols],
-  install_dir : usrbin_exec_dir,
-  install : true)
-if not is_disabler(exe)
-  exes += exe
-  manadocs += ['misc-utils/lsfd.1.adoc']
-endif
+#exe = executable(
+#  'lsfd',
+#  lsfd_sources,
+#  include_directories : includes,
+#  link_with : [lib_common,
+#               lib_smartcols],
+#  install_dir : usrbin_exec_dir,
+#  install : true)
+#if not is_disabler(exe)
+#  exes += exe
+#  manadocs += ['misc-utils/lsfd.1.adoc']
+#endif
 
 exe = executable(
   'uuidgen',
### cat 630-no-uclampset.patch 
# XXX syscall `sched_attr` since Linux 3.14

--- a/meson.build
+++ b/meson.build
@@ -2731,14 +2731,14 @@ exe3 = executable(
   install : opt,
   build_by_default : opt)
 
-exe4 = executable(
-  'uclampset',
-  'schedutils/uclampset.c',
-  include_directories : includes,
-  link_with : lib_common,
-  install_dir : usrbin_exec_dir,
-  install : opt,
-  build_by_default : opt)
+#exe4 = executable(
+#  'uclampset',
+#  'schedutils/uclampset.c',
+#  include_directories : includes,
+#  link_with : lib_common,
+#  install_dir : usrbin_exec_dir,
+#  install : opt,
+#  build_by_default : opt)
 
 if opt and not is_disabler(exe)
   exes += [exe, exe2, exe3, exe4]
