--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -21,17 +21,17 @@
 #undef _PATH_DEFPATH
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/usr/bin"
 #else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/bin:/usr/bin"
 #endif
 
 #undef _PATH_DEFPATH_ROOT
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
 #else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
 #endif
 
 #define	_PATH_HUSHLOGIN		".hushlogin"
@@ -58,11 +58,11 @@
 #define _PATH_TERMCOLORS_DIR	"/etc/" _PATH_TERMCOLORS_DIRNAME
 
 /* login paths */
-#define _PATH_PASSWD		"/etc/passwd"
-#define _PATH_GSHADOW		"/etc/gshadow"
-#define _PATH_GROUP		"/etc/group"
-#define _PATH_SHADOW_PASSWD	"/etc/shadow"
-#define _PATH_SHELLS		"/etc/shells"
+#define _PATH_PASSWD		"/opt/etc/passwd"
+#define _PATH_GSHADOW		"/opt/etc/gshadow"
+#define _PATH_GROUP		"/opt/etc/group"
+#define _PATH_SHADOW_PASSWD	"/opt/etc/shadow"
+#define _PATH_SHELLS		"/opt/etc/shells"
 
 #ifndef _PATH_BTMP
 # define _PATH_BTMP		"/var/log/btmp"
