--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -20,17 +20,17 @@
 #undef _PATH_DEFPATH
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/usr/bin"
 #else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/bin:/bin:/usr/bin"
 #endif
 
 #undef _PATH_DEFPATH_ROOT
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
 #else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/opt/sbin:/opt/bin:/opt/usr/sbin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
 #endif
 
 #define	_PATH_HUSHLOGIN		".hushlogin"
@@ -39,7 +39,7 @@
 #define _PATH_NOLOGIN_TXT	"/etc/nologin.txt"
 
 #ifndef _PATH_MAILDIR
-# define _PATH_MAILDIR		"/var/spool/mail"
+# define _PATH_MAILDIR		"/opt/var/spool/mail"
 #endif
 #define	_PATH_MOTDFILE		"/usr/share/misc/motd:/run/motd:/etc/motd"
 #ifndef _PATH_NOLOGIN
@@ -54,17 +54,17 @@
 #define _PATH_POWEROFF		"/sbin/poweroff"
 
 #define _PATH_TERMCOLORS_DIRNAME "terminal-colors.d"
-#define _PATH_TERMCOLORS_DIR	"/etc/" _PATH_TERMCOLORS_DIRNAME
+#define _PATH_TERMCOLORS_DIR	"/opt/etc/" _PATH_TERMCOLORS_DIRNAME
 
 /* login paths */
-#define _PATH_PASSWD		"/etc/passwd"
-#define _PATH_GSHADOW		"/etc/gshadow"
-#define _PATH_GROUP		"/etc/group"
-#define _PATH_SHADOW_PASSWD	"/etc/shadow"
-#define _PATH_SHELLS		"/etc/shells"
+#define _PATH_PASSWD		"/opt/etc/passwd"
+#define _PATH_GSHADOW		"/opt/etc/gshadow"
+#define _PATH_GROUP		"/opt/etc/group"
+#define _PATH_SHADOW_PASSWD	"/opt/etc/shadow"
+#define _PATH_SHELLS		"/opt/etc/shells"
 
 #ifndef _PATH_TMP
-# define _PATH_TMP		"/tmp/"
+# define _PATH_TMP		"/opt/tmp/"
 #endif
 
 #ifndef _PATH_BTMP
@@ -77,8 +77,8 @@
 #define _PATH_ISSUE		"/etc/" _PATH_ISSUE_FILENAME
 #define _PATH_ISSUEDIR		"/etc/" _PATH_ISSUE_DIRNAME
 
-#define _PATH_OS_RELEASE_ETC	"/etc/os-release"
-#define _PATH_OS_RELEASE_USR	"/usr/lib/os-release"
+#define _PATH_OS_RELEASE_ETC	"/opt/etc/os-release"
+#define _PATH_OS_RELEASE_USR	"/opt/usr/lib/os-release"
 #define _PATH_NUMLOCK_ON	_PATH_RUNSTATEDIR "/numlock-on"
 #define _PATH_LOGINDEFS		"/etc/login.defs"
 
