# XXX <linux/nsfs.h> since Linux 4.9

--- a/config.h.in
+++ b/config.h.in
@@ -360,7 +360,8 @@
 #undef HAVE_LINUX_NET_NAMESPACE_H
 
 /* Define to 1 if you have the <linux/nsfs.h> header file. */
-#undef HAVE_LINUX_NSFS_H
+/* NS_GET_NSTYPE NS_GET_OWNER_UID since Linux 4.11 */
+/* #undef HAVE_LINUX_NSFS_H */
 
 /* Define to 1 if you have the <linux/pr.h> header file. */
 #undef HAVE_LINUX_PR_H
--- a/meson.build
+++ b/meson.build
@@ -195,7 +195,6 @@ headers = '''
         linux/landlock.h
         linux/kcmp.h
         linux/net_namespace.h
-        linux/nsfs.h
         linux/mount.h
         linux/pr.h
         linux/securebits.h
