# XXX <linux/blkzoned.h> since Linux 4.10

--- a/meson.build
+++ b/meson.build
@@ -161,7 +161,6 @@ headers = '''
         asm-generic/fcntl.h
         asm/fcntl.h
         asm/io.h
-        linux/blkzoned.h
         linux/capability.h
         linux/cdrom.h
         linux/compiler.h
@@ -1444,16 +1443,16 @@ exes += exe
 manadocs += ['sys-utils/blkdiscard.8.adoc']
 bashcompletions += ['blkdiscard']
 
-exe = executable(
-  'blkzone',
-  blkzone_sources,
-  include_directories : includes,
-  link_with : [lib_common],
-  install_dir : sbindir,
-  install : true)
-exes += exe
-manadocs += ['sys-utils/blkzone.8.adoc']
-bashcompletions += ['blkzone']
+#exe = executable(
+#  'blkzone',
+#  blkzone_sources,
+#  include_directories : includes,
+#  link_with : [lib_common],
+#  install_dir : sbindir,
+#  install : true)
+#exes += exe
+#manadocs += ['sys-utils/blkzone.8.adoc']
+#bashcompletions += ['blkzone']
 
 exe = executable(
   'blkpr',
