# XXX <linux/kcmp.h> since Linux 3.5 + missing hdr for aarch64

--- a/meson.build
+++ b/meson.build
@@ -2447,18 +2447,18 @@ if not is_disabler(exe)
   bashcompletions += ['lsblk']
 endif
 
-exe = executable(
-  'lsfd',
-  lsfd_sources,
-  include_directories : includes,
-  link_with : [lib_common,
-               lib_smartcols],
-  install_dir : usrbin_exec_dir,
-  install : true)
-if not is_disabler(exe)
-  exes += exe
-  manadocs += ['misc-utils/lsfd.1.adoc']
-endif
+#exe = executable(
+#  'lsfd',
+#  lsfd_sources,
+#  include_directories : includes,
+#  link_with : [lib_common,
+#               lib_smartcols],
+#  install_dir : usrbin_exec_dir,
+#  install : true)
+#if not is_disabler(exe)
+#  exes += exe
+#  manadocs += ['misc-utils/lsfd.1.adoc']
+#endif
 
 exe = executable(
   'uuidgen',
