# XXX <linux/pr.h> since Linux 4.4

--- a/meson.build
+++ b/meson.build
@@ -1454,15 +1454,15 @@ bashcompletions += ['blkdiscard']
 #manadocs += ['sys-utils/blkzone.8.adoc']
 #bashcompletions += ['blkzone']
 
-exe = executable(
-  'blkpr',
-  blkpr_sources,
-  include_directories : includes,
-  link_with : [lib_common],
-  install_dir : sbindir,
-  install : true)
-exes += exe
-manadocs += ['sys-utils/blkpr.8.adoc']
+#exe = executable(
+#  'blkpr',
+#  blkpr_sources,
+#  include_directories : includes,
+#  link_with : [lib_common],
+#  install_dir : sbindir,
+#  install : true)
+#exes += exe
+#manadocs += ['sys-utils/blkpr.8.adoc']
 
 exe = executable(
   'ldattach',
