# XXX pidfd_open since Linux 5.3

--- a/misc-utils/meson.build
+++ b/misc-utils/meson.build
@@ -193,9 +193,9 @@ fadvise_sources = files(
   'fadvise.c',
 )
 
-waitpid_sources = files(
-  'waitpid.c',
-)
+#waitpid_sources = files(
+#  'waitpid.c',
+#)
 
 lsclocks_sources = files(
   'lsclocks.c',
--- a/meson.build
+++ b/meson.build
@@ -3085,20 +3085,20 @@ if opt and not is_disabler(exe)
   bashcompletions += ['fadvise']
 endif
 
-if LINUX and conf.get('HAVE_PIDFD_OPEN').to_string() == '1'
-  exe = executable(
-    'waitpid',
-    waitpid_sources,
-    include_directories : includes,
-    link_with : [lib_common],
-    install_dir : usrbin_exec_dir,
-    install : true)
-  if not is_disabler(exe)
-    exes += exe
-    manadocs += ['misc-utils/waitpid.1.adoc']
-    bashcompletions += ['waitpid']
-  endif
-endif
+#if LINUX and conf.get('HAVE_PIDFD_OPEN').to_string() == '1'
+#  exe = executable(
+#    'waitpid',
+#    waitpid_sources,
+#    include_directories : includes,
+#    link_with : [lib_common],
+#    install_dir : usrbin_exec_dir,
+#    install : true)
+#  if not is_disabler(exe)
+#    exes += exe
+#    manadocs += ['misc-utils/waitpid.1.adoc']
+#    bashcompletions += ['waitpid']
+#  endif
+#endif
 
 syscalls_h = custom_target('syscalls.h',
   input : 'tools/all_syscalls',
