# DP: Proposed backport for PR libstdc++/61841

--- a/libstdc++-v3/include/std/thread
+++ b/libstdc++-v3/include/std/thread
@@ -132,6 +132,9 @@
       explicit 
       thread(_Callable&& __f, _Args&&... __args)
       {
+#ifdef GTHR_ACTIVE_PROXY
+	__asm ("" : : "r" (&pthread_create));
+#endif
         _M_start_thread(_M_make_routine(std::__bind_simple(
                 std::forward<_Callable>(__f),
                 std::forward<_Args>(__args)...)));
