From 3e8586eb3509e2f0d6dfb74be8f89a30b06b56e9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 18 Mar 2015 00:33:03 +0000
Subject: [PATCH 16/24] timezone: re-written tzselect as posix sh

To avoid the bash dependency.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 timezone/Makefile     | 2 +-
 timezone/tzselect.ksh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/timezone/Makefile b/timezone/Makefile
index 99566cb..b6e757e 100644
--- a/timezone/Makefile
+++ b/timezone/Makefile
@@ -120,7 +120,7 @@ $(testdata)/XT%: testdata/XT%
 	cp $< $@
 
 $(objpfx)tzselect: tzselect.ksh $(common-objpfx)config.make
-	sed -e 's|/bin/bash|$(BASH)|' \
+	sed -e 's|/bin/bash|/bin/sh|' \
 	    -e 's|TZDIR=[^}]*|TZDIR=$(zonedir)|' \
 	    -e '/TZVERSION=/s|see_Makefile|"$(version)"|' \
 	    -e '/PKGVERSION=/s|=.*|="$(PKGVERSION)"|' \
diff --git a/timezone/tzselect.ksh b/timezone/tzselect.ksh
index 9d70691..25f45a8 100755
--- a/timezone/tzselect.ksh
+++ b/timezone/tzselect.ksh
@@ -35,7 +35,7 @@

 # Specify default values for environment variables if they are unset.
 : ${AWK=awk}
-: ${TZDIR=`pwd`}
+: ${TZDIR=$(pwd)}

 # Output one argument as-is to standard output.
 # Safer than 'echo', which can mishandle '\' or leading '-'.

