--- a/Makeconfig
+++ b/Makeconfig
@@ -270,7 +270,7 @@ inst_vardbdir = $(install_root)$(vardbdi
 # other than there, so the zoneinfo directory contains only universal data,
 # localizing the configuration data elsewhere.
 ifndef localtime-file
-localtime-file = $(sysconfdir)/localtime
+localtime-file = /opt/etc/localtime
 endif
 
 # What to use for leap second specifications in compiling the default
