--- a/src/crypto/x509/root_linux.go
+++ b/src/crypto/x509/root_linux.go
@@ -8,6 +8,7 @@ import "internal/goos"
 
 // Possible certificate files; stop after finding one.
 var certFiles = []string{
+	"/opt/etc/ssl/certs/ca-certificates.crt",            // Entware
 	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
 	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
 	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
@@ -18,6 +19,7 @@ var certFiles = []string{
 
 // Possible directories with certificate files; all will be read.
 var certDirectories = []string{
+	"/opt/etc/ssl/certs", // Entware
 	"/etc/ssl/certs",     // SLES10/SLES11, https://golang.org/issue/12139
 	"/etc/pki/tls/certs", // Fedora/RHEL
 }
--- a/src/net/nss.go
+++ b/src/net/nss.go
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	nssConfigPath = "/etc/nsswitch.conf"
+	nssConfigPath = "/opt/etc/nsswitch.conf"
 )
 
 var nssConfig nsswitchConfig
@@ -40,7 +40,7 @@ func getSystemNSS() *nssConf {
 
 // init initializes conf and is only called via conf.initOnce.
 func (conf *nsswitchConfig) init() {
-	conf.nssConf = parseNSSConfFile("/etc/nsswitch.conf")
+	conf.nssConf = parseNSSConfFile("/opt/etc/nsswitch.conf")
 	conf.lastChecked = time.Now()
 	conf.ch = make(chan struct{}, 1)
 }
--- a/src/time/zoneinfo_unix.go
+++ b/src/time/zoneinfo_unix.go
@@ -19,6 +19,7 @@ import (
 // /usr/share/lib/zoneinfo, IRIX 6 has /usr/lib/locale/TZ,
 // NixOS has /etc/zoneinfo.
 var platformZoneSources = []string{
+	"/opt/share/zoneinfo/",		// Entware
 	"/usr/share/zoneinfo/",
 	"/usr/share/lib/zoneinfo/",
 	"/usr/lib/locale/TZ/",
@@ -36,7 +37,7 @@ func initLocal() {
 	tz, ok := syscall.Getenv("TZ")
 	switch {
 	case !ok:
-		z, err := loadLocation("localtime", []string{"/etc"})
+		z, err := loadLocation("localtime", []string{"/opt/etc"})
 		if err == nil {
 			localLoc = *z
 			localLoc.name = "Local"
@@ -49,7 +50,7 @@ func initLocal() {
 		if tz != "" && tz[0] == '/' {
 			if z, err := loadLocation(tz, []string{""}); err == nil {
 				localLoc = *z
-				if tz == "/etc/localtime" {
+				if tz == "/opt/etc/localtime" {
 					localLoc.name = "Local"
 				} else {
 					localLoc.name = tz
