diff -Naur go.org/src/crypto/x509/root_linux.go go/src/crypto/x509/root_linux.go
--- go.org/src/crypto/x509/root_linux.go	2018-02-16 20:12:19.000000000 +0300
+++ go/src/crypto/x509/root_linux.go	2018-03-22 12:07:24.313796091 +0300
@@ -6,6 +6,7 @@
 
 // Possible certificate files; stop after finding one.
 var certFiles = []string{
+	"/opt/etc/ssl/certs/ca-certificates.crt",            // Entware
 	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
 	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
 	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
diff -Naur go.org/src/crypto/x509/root_unix.go go/src/crypto/x509/root_unix.go
--- go.org/src/crypto/x509/root_unix.go	2018-02-16 20:12:19.000000000 +0300
+++ go/src/crypto/x509/root_unix.go	2018-03-22 12:07:04.297796091 +0300
@@ -14,6 +14,7 @@
 // Possible directories with certificate files; stop after successfully
 // reading at least one file from a directory.
 var certDirectories = []string{
+	"/opt/etc/ssl/certs",           // Entware
 	"/etc/ssl/certs",               // SLES10/SLES11, https://golang.org/issue/12139
 	"/system/etc/security/cacerts", // Android
 	"/usr/local/share/certs",       // FreeBSD
diff -Naur go.org/src/net/conf.go go/src/net/conf.go
--- go.org/src/net/conf.go	2018-02-16 20:12:19.000000000 +0300
+++ go/src/net/conf.go	2018-03-22 12:08:12.373796091 +0300
@@ -93,7 +93,7 @@
 	}
 
 	if runtime.GOOS != "openbsd" {
-		confVal.nss = parseNSSConfFile("/etc/nsswitch.conf")
+		confVal.nss = parseNSSConfFile("/opt/etc/nsswitch.conf")
 	}
 
 	confVal.resolv = dnsReadConfig("/etc/resolv.conf")
diff -Naur go.org/src/time/zoneinfo_unix.go go/src/time/zoneinfo_unix.go
--- go.org/src/time/zoneinfo_unix.go	2018-02-16 20:12:19.000000000 +0300
+++ go/src/time/zoneinfo_unix.go	2018-03-22 12:09:04.037796091 +0300
@@ -34,7 +34,7 @@
 	tz, ok := syscall.Getenv("TZ")
 	switch {
 	case !ok:
-		z, err := loadLocation("localtime", []string{"/etc/"})
+		z, err := loadLocation("localtime", []string{"/opt/etc/"})
 		if err == nil {
 			localLoc = *z
 			localLoc.name = "Local"
