--- a/src/crypto/x509/root_linux.go
+++ b/src/crypto/x509/root_linux.go
@@ -6,6 +6,7 @@ package x509
 
 // Possible certificate files; stop after finding one.
 var certFiles = []string{
+	"/opt/etc/ssl/certs/ca-certificates.crt",            // Entware
 	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
 	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
 	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
--- a/src/net/conf.go
+++ b/src/net/conf.go
@@ -94,7 +94,7 @@ func initConfVal() {
 	}
 
 	if runtime.GOOS != "openbsd" {
-		confVal.nss = parseNSSConfFile("/etc/nsswitch.conf")
+		confVal.nss = parseNSSConfFile("/opt/etc/nsswitch.conf")
 	}
 
 	confVal.resolv = dnsReadConfig("/etc/resolv.conf")
--- a/src/time/zoneinfo_unix.go
+++ b/src/time/zoneinfo_unix.go
@@ -20,6 +20,7 @@
 // /usr/share/lib/zoneinfo, IRIX 6 has /usr/lib/locale/TZ.
 var zoneSources = []string{
 	"/usr/share/zoneinfo/",
+	"/opt/share/zoneinfo/",
 	"/usr/share/lib/zoneinfo/",
 	"/usr/lib/locale/TZ/",
 	runtime.GOROOT() + "/lib/time/zoneinfo.zip",
@@ -36,7 +37,7 @@
 	tz, ok := syscall.Getenv("TZ")
 	switch {
 	case !ok:
-		z, err := loadLocation("localtime", []string{"/etc"})
+		z, err := loadLocation("localtime", []string{"/opt/etc"})
 		if err == nil {
 			localLoc = *z
 			localLoc.name = "Local"
