--- a/nss/nss_compat/compat-grp.c
+++ b/nss/nss_compat/compat-grp.c
@@ -106,7 +106,7 @@
 
   if (ent->stream == NULL)
     {
-      ent->stream = fopen ("/etc/group", "rme");
+      ent->stream = fopen ("/opt/etc/group", "rme");
 
       if (ent->stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
--- a/nss/nss_compat/compat-initgroups.c
+++ b/nss/nss_compat/compat-initgroups.c
@@ -121,7 +121,7 @@
   else
     ent->blacklist.current = 0;
 
-  ent->stream = fopen ("/etc/group", "rme");
+  ent->stream = fopen ("/opt/etc/group", "rme");
 
   if (ent->stream == NULL)
     status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
--- a/nss/nss_compat/compat-pwd.c
+++ b/nss/nss_compat/compat-pwd.c
@@ -221,7 +221,7 @@
 
   if (ent->stream == NULL)
     {
-      ent->stream = fopen ("/etc/passwd", "rme");
+      ent->stream = fopen ("/opt/etc/passwd", "rme");
 
       if (ent->stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
--- a/nss/nss_compat/compat-spwd.c
+++ b/nss/nss_compat/compat-spwd.c
@@ -177,7 +177,7 @@
 
   if (ent->stream == NULL)
     {
-      ent->stream = fopen ("/etc/shadow", "rme");
+      ent->stream = fopen ("/opt/etc/shadow", "rme");
 
       if (ent->stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
--- a/nss/nss_files/files-grp.c
+++ b/nss/nss_files/files-grp.c
@@ -21,6 +21,7 @@
 #define STRUCTURE	group
 #define ENTNAME		grent
 #define DATABASE	"group"
+#define DATABASE_PREFIX
 struct grent_data {};
 
 /* Our parser function is already defined in fgetgrent.c, so use that.
--- a/nss/nss_files/files-init.c
+++ b/nss/nss_files/files-init.c
@@ -21,10 +21,10 @@
 #include <string.h>
 #include <nscd/nscd.h>
 
-#define PWD_FILENAME "/etc/passwd"
+#define PWD_FILENAME "/opt/etc/passwd"
 define_traced_file (pwd, PWD_FILENAME);
 
-#define GRP_FILENAME "/etc/group"
+#define GRP_FILENAME "/opt/etc/group"
 define_traced_file (grp, GRP_FILENAME);
 
 #define HST_FILENAME "/etc/hosts"
--- a/nss/nss_files/files-initgroups.c
+++ b/nss/nss_files/files-initgroups.c
@@ -31,7 +31,7 @@
 			   long int *size, gid_t **groupsp, long int limit,
 			   int *errnop)
 {
-  FILE *stream = fopen ("/etc/group", "rce");
+  FILE *stream = fopen ("/opt/etc/group", "rce");
   if (stream == NULL)
     {
       *errnop = errno;
--- a/nss/nss_files/files-pwd.c
+++ b/nss/nss_files/files-pwd.c
@@ -21,6 +21,7 @@
 #define STRUCTURE	passwd
 #define ENTNAME		pwent
 #define DATABASE	"passwd"
+#define DATABASE_PREFIX
 struct pwent_data {};
 
 /* Our parser function is already defined in fgetpwent_r.c, so use that
--- a/nss/nss_files/files-sgrp.c
+++ b/nss/nss_files/files-sgrp.c
@@ -21,6 +21,7 @@
 #define STRUCTURE	sgrp
 #define ENTNAME		sgent
 #define DATABASE	"gshadow"
+#define DATABASE_PREFIX
 struct sgent_data {};
 
 /* Our parser function is already defined in sgetspent_r.c, so use that
--- a/nss/nss_files/files-spwd.c
+++ b/nss/nss_files/files-spwd.c
@@ -21,6 +21,7 @@
 #define STRUCTURE	spwd
 #define ENTNAME		spent
 #define DATABASE	"shadow"
+#define DATABASE_PREFIX
 struct spent_data {};
 
 /* Our parser function is already defined in sgetspent_r.c, so use that
--- a/nss/nss_files/files-XXX.c
+++ b/nss/nss_files/files-XXX.c
@@ -38,7 +38,11 @@
 
 #define ENTNAME_r	CONCAT(ENTNAME,_r)
 
+#ifdef DATABASE_PREFIX
+#define DATAFILE	"/opt/etc/" DATABASE
+#else
 #define DATAFILE	"/etc/" DATABASE
+#endif
 
 #ifdef NEED_H_ERRNO
 # include <netdb.h>
--- a/shadow/lckpwdf.c
+++ b/shadow/lckpwdf.c
@@ -30,7 +30,7 @@
 
 
 /* Name of the lock file.  */
-#define PWD_LOCKFILE "/etc/.pwd.lock"
+#define PWD_LOCKFILE "/opt/etc/.pwd.lock"
 
 /* How long to wait for getting the lock before returning with an
    error.  */
--- a/sysdeps/generic/paths.h
+++ b/sysdeps/generic/paths.h
@@ -44,7 +44,7 @@
 #define	_PATH_DEVDB	"/var/run/dev.db"
 #define	_PATH_DEVNULL	"/dev/null"
 #define	_PATH_DRUM	"/dev/drum"
-#define	_PATH_GSHADOW	"/etc/gshadow"
+#define	_PATH_GSHADOW	"/opt/etc/gshadow"
 #define	_PATH_KMEM	"/dev/kmem"
 #define	_PATH_LASTLOG	"/var/log/lastlog"
 #define	_PATH_MAILDIR	"/var/mail"
@@ -56,8 +56,8 @@
 #define	_PATH_PRESERVE	"/var/lib"
 #define	_PATH_RWHODIR	"/var/spool/rwho"
 #define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
-#define	_PATH_SHADOW	"/etc/shadow"
-#define	_PATH_SHELLS	"/etc/shells"
+#define	_PATH_SHADOW	"/opt/etc/shadow"
+#define	_PATH_SHELLS	"/opt/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
 #define	_PATH_UNIX	"/vmunix"
 #define	_PATH_UTMP	"/var/run/utmp"
--- a/sysdeps/unix/sysv/linux/paths.h
+++ b/sysdeps/unix/sysv/linux/paths.h
@@ -44,7 +44,7 @@
 #define	_PATH_DEVDB	"/var/run/dev.db"
 #define	_PATH_DEVNULL	"/dev/null"
 #define	_PATH_DRUM	"/dev/drum"
-#define	_PATH_GSHADOW	"/etc/gshadow"
+#define	_PATH_GSHADOW	"/opt/etc/gshadow"
 #define	_PATH_KLOG	"/proc/kmsg"
 #define	_PATH_KMEM	"/dev/kmem"
 #define	_PATH_LASTLOG	"/var/log/lastlog"
@@ -57,8 +57,8 @@
 #define	_PATH_PRESERVE	"/var/lib"
 #define	_PATH_RWHODIR	"/var/spool/rwho"
 #define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
-#define	_PATH_SHADOW	"/etc/shadow"
-#define	_PATH_SHELLS	"/etc/shells"
+#define	_PATH_SHADOW	"/opt/etc/shadow"
+#define	_PATH_SHELLS	"/opt/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
 #define	_PATH_UNIX	"/boot/vmlinux"
 #define	_PATH_UTMP	"/var/run/utmp"
