--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ OPTION(BUILD_LUA "build Lua plugin" ON)
 OPTION(BUILD_EXAMPLES "build examples" ON)
 
 SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "")
-SET(UBUS_UNIX_SOCKET "/var/run/ubus/ubus.sock")
+SET(UBUS_UNIX_SOCKET "/opt/var/run/ubus/ubus.sock")
 SET(UBUS_MAX_MSGLEN 1048576)
 
 ADD_DEFINITIONS( -DUBUS_UNIX_SOCKET="${UBUS_UNIX_SOCKET}")
--- a/ubusd_acl.c
+++ b/ubusd_acl.c
@@ -72,7 +72,7 @@ struct ubusd_acl_file {
 	int ok;
 };
 
-const char *ubusd_acl_dir = "/usr/share/acl.d";
+const char *ubusd_acl_dir = "/opt/share/acl.d";
 static struct blob_buf bbuf;
 static struct avl_tree ubusd_acls;
 static int ubusd_acl_seq;
