--- a/locale/Makefile
+++ b/locale/Makefile
@@ -76,6 +76,12 @@
 	$(PERL) gen-translit.pl < $< > $@.tmp
 	mv -f $@.tmp $@
 
+localedir = /opt/usr/lib/locale
+i18ndir = /opt/usr/share/i18n
+rtld-LDFLAGS = -Wl,--dynamic-linker=/opt/lib/ld-linux.so.3
+LDFLAGS-locale = -Wl,-rpath,/opt/lib
+LDFLAGS-localedef = -Wl,-rpath,/opt/lib
+
 localepath = "$(localedir):$(i18ndir)"
 
 # -Iprograms doesn't really belong here, but this gets it at the head
