--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 # Makefile
 #
 
-CERTSDIR = /usr/share/ca-certificates
+CERTSDIR = /opt/share/ca-certificates
 SUBDIRS = mozilla
 
 all:
--- a/sbin/Makefile
+++ b/sbin/Makefile
@@ -3,7 +3,7 @@
 #
 #
 
-SBINDIR = /usr/sbin
+SBINDIR = /opt/sbin
 
 all:
 
--- a/sbin/update-ca-certificates
+++ b/sbin/update-ca-certificates
@@ -24,12 +24,12 @@
 verbose=0
 fresh=0
 default=0
-CERTSCONF=/etc/ca-certificates.conf
-CERTSDIR=/usr/share/ca-certificates
-LOCALCERTSDIR=/usr/local/share/ca-certificates
+CERTSCONF=/opt/etc/ca-certificates.conf
+CERTSDIR=/opt/share/ca-certificates
+LOCALCERTSDIR=/opt/share/ca-certificates
 CERTBUNDLE=ca-certificates.crt
-ETCCERTSDIR=/etc/ssl/certs
-HOOKSDIR=/etc/ca-certificates/update.d
+ETCCERTSDIR=/opt/etc/ssl/certs
+HOOKSDIR=/opt/etc/ca-certificates/update.d
 
 while [ $# -gt 0 ];
 do
