--- ppl-1.1-vanilla/src/Congruence_System_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Congruence_System_defs.hh	2014-04-20 20:40:41.839919465 +0200
@@ -249,7 +249,7 @@
   class const_iterator
     : public std::iterator<std::forward_iterator_tag,
                            Congruence,
-                           ptrdiff_t,
+                           std::ptrdiff_t,
                            const Congruence*,
                            const Congruence&> {
   public:
--- ppl-1.1-vanilla/src/Constraint_System_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Constraint_System_defs.hh	2014-04-20 20:35:12.263921783 +0200
@@ -609,7 +609,7 @@
 class Parma_Polyhedra_Library::Constraint_System_const_iterator
   : public std::iterator<std::forward_iterator_tag,
                          Constraint,
-                         ptrdiff_t,
+                         std::ptrdiff_t,
                          const Constraint*,
                          const Constraint&> {
 public:
--- ppl-1.1-vanilla/src/CO_Tree.cc	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/CO_Tree.cc	2014-04-20 20:25:35.311925841 +0200
@@ -954,7 +954,7 @@
     --subtree_size;
   }
 
-  const ptrdiff_t distance = first_unused_index - indexes;
+  const std::ptrdiff_t distance = first_unused_index - indexes;
   PPL_ASSERT(distance >= 0);
   return static_cast<dimension_type>(distance);
 }
--- ppl-1.1-vanilla/src/CO_Tree_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/CO_Tree_defs.hh	2014-04-20 20:10:58.843932005 +0200
@@ -159,7 +159,7 @@
 
     typedef std::bidirectional_iterator_tag iterator_category;
     typedef const data_type value_type;
-    typedef ptrdiff_t difference_type;
+    typedef std::ptrdiff_t difference_type;
     typedef value_type* pointer;
     typedef data_type_const_reference reference;
 
@@ -314,7 +314,7 @@
 
     typedef std::bidirectional_iterator_tag iterator_category;
     typedef data_type value_type;
-    typedef ptrdiff_t difference_type;
+    typedef std::ptrdiff_t difference_type;
     typedef value_type* pointer;
     typedef value_type& reference;
 
--- ppl-1.1-vanilla/src/CO_Tree_inlines.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/CO_Tree_inlines.hh	2014-04-20 20:13:40.099930871 +0200
@@ -31,7 +31,7 @@
   PPL_ASSERT(itr.current_index != 0);
   PPL_ASSERT(itr.current_index >= indexes + 1);
   PPL_ASSERT(itr.current_index <= indexes + reserved_size);
-  const ptrdiff_t index = itr.current_index - indexes;
+  const std::ptrdiff_t index = itr.current_index - indexes;
   return static_cast<dimension_type>(index);
 }
 
@@ -40,7 +40,7 @@
   PPL_ASSERT(itr.current_index != 0);
   PPL_ASSERT(itr.current_index >= indexes + 1);
   PPL_ASSERT(itr.current_index <= indexes + reserved_size);
-  const ptrdiff_t index = itr.current_index - indexes;
+  const std::ptrdiff_t index = itr.current_index - indexes;
   return static_cast<dimension_type>(index);
 }
 
@@ -772,7 +772,7 @@
   p -= (offset - 1);
   while (*p == unused_index)
     ++p;
-  const ptrdiff_t distance = p - tree.indexes;
+  const std::ptrdiff_t distance = p - tree.indexes;
   PPL_ASSERT(distance >= 0);
   i = static_cast<dimension_type>(distance);
   offset = least_significant_one_mask(i);
@@ -787,7 +787,7 @@
   p += (offset - 1);
   while (*p == unused_index)
     --p;
-  const ptrdiff_t distance = p - tree.indexes;
+  const std::ptrdiff_t distance = p - tree.indexes;
   PPL_ASSERT(distance >= 0);
   i = static_cast<dimension_type>(distance);
   offset = least_significant_one_mask(i);
--- ppl-1.1-vanilla/src/Dense_Row_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Dense_Row_defs.hh	2014-04-20 20:09:14.611932740 +0200
@@ -433,7 +433,7 @@
 
   typedef std::bidirectional_iterator_tag iterator_category;
   typedef Coefficient value_type;
-  typedef ptrdiff_t difference_type;
+  typedef std::ptrdiff_t difference_type;
   typedef value_type* pointer;
   typedef value_type& reference;
 
@@ -474,7 +474,7 @@
 public:
 
   typedef const Coefficient value_type;
-  typedef ptrdiff_t difference_type;
+  typedef std::ptrdiff_t difference_type;
   typedef value_type* pointer;
   typedef Coefficient_traits::const_reference reference;
 
--- ppl-1.1-vanilla/src/Generator_System_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Generator_System_defs.hh	2014-04-20 20:45:38.327917382 +0200
@@ -679,7 +679,7 @@
 class Parma_Polyhedra_Library::Generator_System_const_iterator
   : public std::iterator<std::forward_iterator_tag,
         Generator,
-        ptrdiff_t,
+        std::ptrdiff_t,
         const Generator*,
         const Generator&> {
 public:
--- ppl-1.1-vanilla/src/Grid_Generator_System_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Grid_Generator_System_defs.hh	2014-04-20 20:52:48.503914356 +0200
@@ -277,7 +277,7 @@
   class const_iterator
     : public std::iterator<std::forward_iterator_tag,
                            Grid_Generator,
-                           ptrdiff_t,
+                           std::ptrdiff_t,
                            const Grid_Generator*,
                            const Grid_Generator&> {
   public:
--- ppl-1.1-vanilla/src/Linear_Expression_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Linear_Expression_defs.hh	2014-04-20 20:19:47.831928285 +0200
@@ -381,7 +381,7 @@
   public:
     typedef std::bidirectional_iterator_tag iterator_category;
     typedef const Coefficient value_type;
-    typedef ptrdiff_t difference_type;
+    typedef std::ptrdiff_t difference_type;
     typedef value_type* pointer;
     typedef Coefficient_traits::const_reference reference;
 
--- ppl-1.1-vanilla/src/Linear_Expression_Interface_defs.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1-fixed-gcc-4.9/src/Linear_Expression_Interface_defs.hh	2014-04-20 20:16:01.107929880 +0200
@@ -65,7 +65,7 @@
   public:
     typedef std::bidirectional_iterator_tag iterator_category;
     typedef const Coefficient value_type;
-    typedef ptrdiff_t difference_type;
+    typedef std::ptrdiff_t difference_type;
     typedef value_type* pointer;
     typedef Coefficient_traits::const_reference reference;
 

