--- a/src/default_options.h
+++ b/src/default_options.h
@@ -23,10 +23,10 @@ IMPORTANT: Some options will require "ma
 /* Default hostkey paths - these can be specified on the command line.
  * Homedir is prepended if path begins with ~/
  */
-#define DSS_PRIV_FILENAME "/etc/dropbear/dropbear_dss_host_key"
-#define RSA_PRIV_FILENAME "/etc/dropbear/dropbear_rsa_host_key"
-#define ECDSA_PRIV_FILENAME "/etc/dropbear/dropbear_ecdsa_host_key"
-#define ED25519_PRIV_FILENAME "/etc/dropbear/dropbear_ed25519_host_key"
+#define DSS_PRIV_FILENAME "/opt/etc/dropbear/dropbear_dss_host_key"
+#define RSA_PRIV_FILENAME "/opt/etc/dropbear/dropbear_rsa_host_key"
+#define ECDSA_PRIV_FILENAME "/opt/etc/dropbear/dropbear_ecdsa_host_key"
+#define ED25519_PRIV_FILENAME "/opt/etc/dropbear/dropbear_ed25519_host_key"
 
 /* Set NON_INETD_MODE if you require daemon functionality (ie Dropbear listens
  * on chosen ports and keeps accepting connections. This is the default.
@@ -223,7 +223,7 @@ group1 in Dropbear server too */
 
 /* Whether to print the message of the day (MOTD). */
 #define DO_MOTD 1
-#define MOTD_FILENAME "/etc/motd"
+#define MOTD_FILENAME "/opt/etc/motd"
 #define MOTD_MAXSIZE 2000
 
 /* Authentication Types - at least one required.
@@ -282,7 +282,7 @@ group1 in Dropbear server too */
 
 /* Set this to use PRNGD or EGD instead of /dev/urandom */
 #define DROPBEAR_USE_PRNGD 0
-#define DROPBEAR_PRNGD_SOCKET "/var/run/dropbear-rng"
+#define DROPBEAR_PRNGD_SOCKET "/opt/var/run/dropbear-rng"
 
 /* Specify the number of clients we will allow to be connected but
  * not yet authenticated. After this limit, connections are rejected */
@@ -307,7 +307,7 @@ group1 in Dropbear server too */
  * scripts etc. This can be overridden with the -P flag.
  * Homedir is prepended if path begins with ~/
  */
-#define DROPBEAR_PIDFILE "/var/run/dropbear.pid"
+#define DROPBEAR_PIDFILE "/opt/var/run/dropbear.pid"
 
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
@@ -320,11 +320,11 @@ group1 in Dropbear server too */
  * Homedir is prepended if path begins with ~/
  */
 #define DROPBEAR_SFTPSERVER 1
-#define SFTPSERVER_PATH "/usr/libexec/sftp-server"
+#define SFTPSERVER_PATH "/opt/libexec/sftp-server"
 
 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
-#define DROPBEAR_PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#define DROPBEAR_PATH_SSH_PROGRAM "/opt/bin/dbclient"
 
 /* Whether to log commands executed by a client. This only logs the
  * (single) command sent to the server, not what a user did in a
@@ -360,7 +360,7 @@ be overridden at runtime with -I. 0 disa
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
-#define DEFAULT_ROOT_PATH "/usr/sbin:/usr/bin:/sbin:/bin"
+#define DEFAULT_PATH "/opt/usr/bin:/opt/bin:/usr/bin:/bin"
+#define DEFAULT_ROOT_PATH "/opt/usr/sbin:/opt/usr/bin:/opt/sbin:/opt/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
 #endif /* DROPBEAR_DEFAULT_OPTIONS_H_ */
--- a/src/sysoptions.h
+++ b/src/sysoptions.h
@@ -98,7 +98,7 @@
   -DCOMPAT_USER_SHELLS='"/bin/sh","/apps/bin/sh","/data/bin/zsh"'
  */
 #ifndef COMPAT_USER_SHELLS
-#define COMPAT_USER_SHELLS "/bin/sh","/bin/csh"
+#define COMPAT_USER_SHELLS "/opt/bin/sh","/bin/sh","/bin/csh"
 #endif
 
 #define DROPBEAR_ESCAPE_CHAR '~'
