--- a/src/svr-authpubkey.c
+++ b/src/svr-authpubkey.c
@@ -78,8 +78,8 @@ static void send_msg_userauth_pk_ok(cons
 		const unsigned char* keyblob, unsigned int keybloblen);
 static int checkfileperm(char * filename);
 
-static const char * const global_authkeys_dir = "/etc/dropbear";
-static const int        n_global_authkeys_dir = 14; /* + 1 extra byte */
+static const char * const global_authkeys_dir = "/opt/etc/dropbear";
+static const int        n_global_authkeys_dir = 18; /* + 1 extra byte */
 static const char * const user_authkeys_dir = ".ssh";
 static const int        n_user_authkeys_dir = 5; /* + 1 extra byte */
 static const char * const authkeys_file = "authorized_keys";
