--- a/net.c
+++ b/net.c
@@ -253,7 +253,7 @@
 	case AF_BLUETOOTH:
 		tprintf("{sco_bdaddr=%02X:%02X:%02X:%02X:%02X:%02X} or "
 			"{rc_bdaddr=%02X:%02X:%02X:%02X:%02X:%02X, rc_channel=%d} or "
-			"{l2_psm=htobs(%d), l2_bdaddr=%02X:%02X:%02X:%02X:%02X:%02X, l2_cid=htobs(%d)} or "
+			"{l2_psm=htobs(%d), l2_bdaddr=%02X:%02X:%02X:%02X:%02X:%02X} or "
 			"{hci_dev=htobs(%d)}",
 			addr->sco.sco_bdaddr.b[0], addr->sco.sco_bdaddr.b[1],
 			addr->sco.sco_bdaddr.b[2], addr->sco.sco_bdaddr.b[3],
@@ -265,7 +265,7 @@
 			btohs(addr->l2.l2_psm), addr->l2.l2_bdaddr.b[0],
 			addr->l2.l2_bdaddr.b[1], addr->l2.l2_bdaddr.b[2],
 			addr->l2.l2_bdaddr.b[3], addr->l2.l2_bdaddr.b[4],
-			addr->l2.l2_bdaddr.b[5], btohs(addr->l2.l2_cid),
+			addr->l2.l2_bdaddr.b[5],
 			btohs(addr->hci.hci_dev));
 		break;
 #endif /* AF_BLUETOOTH && HAVE_BLUETOOTH_BLUETOOTH_H */
