--- a/prctl.c
+++ b/prctl.c
@@ -251,7 +251,7 @@
 
 	case PR_SET_SECCOMP:
 		tprints(", ");
-		printxval_long(seccomp_mode, tcp->u_arg[1],
+		printxval_long(seccomp_modex, tcp->u_arg[1],
 			  "SECCOMP_MODE_???");
 		if (SECCOMP_MODE_STRICT == tcp->u_arg[1])
 			return RVAL_DECODED;
--- a/xlat/seccomp_mode.h
+++ b/xlat/seccomp_mode.h
@@ -16,7 +16,7 @@
 #else
 
 static
-const struct xlat seccomp_mode[] = {
+const struct xlat seccomp_modex[] = {
  XLAT(SECCOMP_MODE_DISABLED),
  XLAT(SECCOMP_MODE_STRICT),
  XLAT(SECCOMP_MODE_FILTER),
