--- a/prctl.c
+++ b/prctl.c
@@ -273,7 +273,7 @@
 
 	case PR_SET_SECCOMP:
 		tprints(", ");
-		printxval64(seccomp_mode, arg2,
+		printxval64(sseccomp_mode, arg2,
 			    "SECCOMP_MODE_???");
 		if (SECCOMP_MODE_STRICT == arg2)
 			return RVAL_DECODED;
--- a/xlat/seccomp_mode.h
+++ b/xlat/seccomp_mode.h
@@ -16,7 +16,7 @@
 #else
 
 static
-const struct xlat seccomp_mode[] = {
+const struct xlat sseccomp_mode[] = {
  XLAT(SECCOMP_MODE_DISABLED),
  XLAT(SECCOMP_MODE_STRICT),
  XLAT(SECCOMP_MODE_FILTER),
